fvm/src/kernel/blocks.rs

```diff
@@ -24,47 +25,50 @@ pub type BlockId = u32;
 const FIRST_ID: BlockId = 1;
 const MAX_BLOCKS: u32 = i32::MAX as u32; // TODO(M2): Limit
 
-/// Codecs allowed by the IPLD subsytem.
-const ALLOWED_CODECS: &[u64; 3] = &[CBOR, DAG_CBOR, IPLD_RAW];
-
 #[derive(Debug, Copy, Clone)]
 pub struct BlockStat {
     pub codec: u64,
     pub size: u32,
 }
 
 #[derive(Debug, Clone)]
-pub struct Block {
+pub struct Block(Rc<BlockInner>);
+#[derive(Debug)]
+struct BlockInner {
     codec: u64,
-    // Unfortunately, we usually start with a vector/boxed buffer. If we used Rc<[u8]>, we'd have to
-    // copy the bytes. So we accept some indirection for reliable performance.
-    #[allow(clippy::redundant_allocation)]
-    data: Rc<Box<[u8]>>,
+    data: Box<[u8]>,
+    links: Box<[Cid]>,
```

Comment:
Ok, you moved the RC pointer into the whole inner struct, I guess the alternative would be to have two RC pointers in this inner block but I guess this is _better_ and would be the same performance

Reply:
This should have better performance as it leads to a single allocation instead of 2. But yeah, that's why I did this.

Well, 3 allocations instead of 4.